hyper/http11/connection.py

```diff
@@ -218,7 +247,8 @@ def get_response(self):
         # https://github.com/Lukasa/hyper/issues/312.
         # Connection options are case-insensitive, while upgrade tokens are
         # case-sensitive: https://github.com/httpwg/http11bis/issues/8.
-        if (response.status == 101 and
+        if (not self._is_connect_request and
```

Comment:
This shouldn't be necessary: as we don't send `Upgrade` headers on a `CONNECT` request, we're not going to get a `101 Switching Protocols` response back. The standard logic is fine.